File: Day 2/solve1.py
def check(n: int) -> bool:
    s = str(n)
    if len(s) % 2 != 0: return False
    check = True
    for j in range(len(s) // 2):
        if s[j] != s[len(s) // 2 + j]: 
            check = False
            break
    return check

with open("input.txt") as f:
    data = f.read().strip().split(",")
    count = 0
    for rng in data:
        a, b = map(int, rng.split("-"))
        for n in range(a, b + 1):
            if not check(n): continue
            count += n
    print(count)

File: Day 6/solve1.go
// Source - https://stackoverflow.com/a
// Posted by Stefan Arentz, modified by community. See post 'Timeline' for change history
// Retrieved 2025-12-18, License - CC BY-SA 4.0

package main

import (
    "bufio"
    "fmt"
    "log"
    "os"
	"strings"
	"strconv"
)

func main() {
    file, err := os.Open("input.txt")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

	var values [][]int;
	ans := 0;
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
		line := scanner.Text();
		
		if line[0] == '*' || line[0] == '+' {
			for i, x := range strings.Fields(line) {
				cur := 1;
				switch x[0] {
				case '*':
					for _, line := range values {cur *= line[i];}
					break;
				case '+':
					cur = 0;
					for _, line := range values {cur += line[i];}
				}
				ans += cur;
			}
			
			fmt.Println(ans);
		} else {
			var data []int;
			for i, x := range strings.Fields(line) {data = append(data, 0); data[i], err = strconv.Atoi(x);}
			values = append(values, data);
		}
    }
}
